CorsixTH/Lua/dialogs/bottom_panel.lua

```diff
@@ -565,34 +577,74 @@ end
 
 function UIBottomPanel:dialogBuildRoom()
   if self.world.user_actions_allowed then
-    local dlg = UIBuildRoom(self.ui)
-    self.ui:setEditRoom(false)
-    self.ui:addWindow(dlg)
-    self.ui:tutorialStep(3, 1, 2)
+    local w = self.ui:getWindow(UIBuildRoom)
+    local fullscreen = self.ui:getWindow(UIFullscreen)
+
+    if w then
```

Comment:
The problem is that each "w" is a different dialog class. i.e. "UIBuildRoom" in "dialogBuildRoom()" and "UIHireStaff" in "dialogHireStaff()". Also, there are different tutorial steps in a few of these dialogs that are different.

Reply:
Since "w" and "fullscreen" variables are used so often, maybe I should define them in file scope and just change them when I need them, to keep from having to create and destroy these variables over and over through the course of a level, as the player might open these dialogs hundreds of times. Would that be enough?